draft-ietf-tls-tls13.md

```diff
@@ -2583,8 +2601,23 @@ the "supported_groups" extension, then the server MUST reply with a
 HelloRetryRequest.  If there is no mutually supported group at all,
 the server MUST NOT negotiate an (EC)DHE cipher suite.
 
-[[TODO: Recommendation about what the client offers.
-Presumably which integer DH groups and which curves.]]
+Clients are RECOMMENDED to send a selection of KeyShareEntry values
+to avoid the round trip time incurred with a HelloRetryRequest.
+In general, it is RECOMMENDED to send at least one group of each security
+level corresponding to the key lengths of the offered bulk ciphers (see
+{{negotiated-groups}}), preferably the highest priority supported groups
+applicable to the highest priority supported cipher suites.
+Groups stated as mandatory to implement for ciphers offered by the client
+are RECOMMENDED to be sent in most, if not all, offers.
+(see {{mti-cipher-suites}}) Clients with prior expectations of what groups
+are likely to be accepted by the server will generally want to include
+those groups in addition to, or instead of, this general set of
+recommendations.
```

Comment:
I'm not sure I think this is the right recommendation, since it would imply
(for instance) that one should send x448 when you are offering ChaCha/Poly (because it has a 256-bit key). That doesn't seem necessary. My sense would be to recommend sending one or two groups plus "prior expectations"